CMakeLists.txt

```diff
@@ -680,7 +680,9 @@ include(FlagsSummary)
 flags_summary()
 message("Attempt to harden executables ......... ${ENABLE_HARDENING}")
 message("Treat compiler warnings as errors ..... ${WERROR}")
-message("Use ccache for compiling .............. ${WITH_CCACHE}")
+if(NOT MSVC)
+  message("Use ccache for compiling .............. ${WITH_CCACHE}")
+endif()
```

Comment:
The [`MSVC`](https://cmake.org/cmake/help/latest/variable/MSVC.html) variable is defined based on the compiler. However, the current ccache settings rely on the [`<LANG>_COMPILER_LAUNCHER`](https://cmake.org/cmake/help/latest/prop_tgt/LANG_COMPILER_LAUNCHER.html) properties, which are considered depending on the generator rather than the compiler.